multimodal/src/autogluon/multimodal/data/process_ovd.py

```diff
@@ -0,0 +1,201 @@
+    @property
+    def prompt_key(self):
+        return f"{self.prefix}_{PROMPT}"
+
+    @property
+    def image_valid_num_key(self):
+        return f"{self.prefix}_{IMAGE_VALID_NUM}"
+
+    @property
+    def image_column_prefix(self):
+        return f"{self.image_key}_{COLUMN}"
+
+    def collate_fn(self, image_column_names: Optional[List] = None, per_gpu_batch_size: Optional[int] = None) -> Dict:
+        """
+        Collate images into a batch. Here it pads images since the image number may
+        vary from sample to sample. Samples with less images will be padded zeros.
+        The valid image numbers of samples will be stacked into a vector.
+        This function will be used when creating Pytorch DataLoader.
+
+        Returns
+        -------
+        A dictionary containing one model's collator function for image data.
+        """
+        fn = {}
+        if self.requires_column_info:
+            return NotImplementedError(
+                f"requires_column_info={self.requires_column_info} not implemented for OVD tasks."
+            )
+
+        fn.update(
+            {
+                self.image_key: PadCollator(pad_val=0),
+                self.prompt_key: ListCollator(),
+                self.image_meta_key: ListCollator(),
+            }
+        )
+        return fn
+
+    def process_one_sample(
+        self,
+        texts_and_images: Dict[str, Union[str, List[str]]],
+        is_training: bool,
+    ) -> Dict:
+        """
+        Read images, process them, and stack them. One sample can have multiple images,
+        resulting in a tensor of (n, 3, size, size), where n <= max_img_num_per_col is the available image number.
+
+        Parameters
+        ----------
+        texts_and_images
+            The input could be image or text (image captions) for open vocabulary detection.
+        is_training
+            Whether to process images in the training mode.
+
+        Returns
+        -------
+        A dictionary containing one sample's images and their number.
+        """
+        ret = {}
+        image_data = {}
+
+        for per_col_name, per_col_content in texts_and_images.items():
+            if is_image_input(per_col_content, IMAGE_PATH):
+                with PIL.Image.open(per_col_content[0]) as img:
+                    image_data[self.image_key] = dict(filename=per_col_content[0], height=img.height, width=img.width)
+                    ret[self.image_meta_key] = [img.width, img.height]
+            elif is_rois_input(per_col_content):
+                raise NotImplementedError(
+                    "Finetuning/Evaluation with ground truth labels are not implemented for OVD yet"
+                )  # TODO
+            else:
+                ret[self.prompt_key] = per_col_content[0]
+        if self.requires_column_info:
+            pass  # TODO
+
+        ret.update(
+            {self.image_key: self.train_processor(image_data) if is_training else self.val_processor(image_data)}
+        )
```

Comment:
The image and text processing are independent. I think no need to put them in one processor.

Reply:
As above, will refactor out text processor in next PR. But have to keep image processing here for later training support.